packages/go/cypher/backend/pgsql/format.go

```diff
@@ -240,207 +240,175 @@ func (s *Emitter) writeWhere(writer io.Writer, whereClause *model.Where) error {
+	var (
+		pathspaceTable = traversalTablePrefix + strconv.Itoa(idx)
+	)
 
-					if _, err := WriteStrings(writer, ".id = "); err != nil {
-						return err
-					}
+	// Each relationship element should be authored as a recursive CTE
+	if _, err := WriteStrings(writer, traversalHeader, pathspaceTable, traversalTableDef); err != nil {
+		return err
+	}
 
-					if err := s.WriteExpression(writer, previousRelationshipPattern.Binding); err != nil {
-						return err
-					}
+	// Author the initial condition
+	switch relationshipPattern.Direction {
+	case graph.DirectionOutbound:
+		if _, err := WriteStrings(writer, "select ", relationshipBinding, ".end_id, 0, false, array[", relationshipBinding, ".id] from edge ", relationshipBinding, " "); err != nil {
+			return err
+		}
 
-					if _, err := WriteStrings(writer, ".end_id "); err != nil {
-						return err
-					}
-				}
+		if nodePattern.Binding != nil {
+			if _, err := WriteStrings(writer, "join node ", nodeBinding, " on ", nodeBinding, ".id = ", relationshipBinding, ".start_id"); err != nil {
+				return err
 			}
-		} else {
-			relationshipPattern, _ := patternElement.AsRelationshipPattern()
-
-			if idx == 0 {
-				if _, err := io.WriteString(writer, pgDriverModel.EdgeTable); err != nil {
-					return nil
-				}
-
-				if _, err := io.WriteString(writer, " as "); err != nil {
-					return nil
-				}
-
-				if err := s.WriteExpression(writer, relationshipPattern.Binding); err != nil {
-					return nil
-				}
-			} else {
-				previousNodePattern, _ := patternElements[idx-1].AsNodePattern()
-
-				if _, err := WriteStrings(writer, " join ", pgDriverModel.EdgeTable, " "); err != nil {
-					return err
-				}
-
-				if err := s.WriteExpression(writer, relationshipPattern.Binding); err != nil {
-					return err
-				}
+		}
 
-				if _, err := WriteStrings(writer, " on "); err != nil {
-					return err
-				}
+	case graph.DirectionInbound:
+		if _, err := WriteStrings(writer, "select ", relationshipBinding, ".start_id, 0, false, array[", relationshipBinding, ".id] from edge ", relationshipBinding, " "); err != nil {
+			return err
+		}
 
-				switch relationshipPattern.Direction {
-				case graph.DirectionOutbound:
-					if err := s.WriteExpression(writer, relationshipPattern.Binding); err != nil {
-						return err
-					}
+		if nodePattern.Binding != nil {
+			if _, err := WriteStrings(writer, "join node ", nodeBinding, " on ", nodeBinding, ".id = ", relationshipBinding, ".end_id"); err != nil {
+				return err
+			}
+		}
 
-					if _, err := WriteStrings(writer, ".start_id = "); err != nil {
-						return err
-					}
+	default:
+		return fmt.Errorf("unsupported direction: %s(%d)", relationshipPattern.Direction, relationshipPattern.Direction)
+	}
 
-					if err := s.WriteExpression(writer, previousNodePattern.Binding); err != nil {
-						return err
-					}
+	return nil
+}
 
-					if _, err := WriteStrings(writer, ".id"); err != nil {
-						return err
-					}
+// []->()
+// []-()
```

Comment:
Without much of an explanation these comments just look out of place. Could you elaborate on how this relates to the method below or remove it?